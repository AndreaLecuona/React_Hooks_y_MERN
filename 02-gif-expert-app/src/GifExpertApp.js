import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ( { defaultCategories = [] } ) => {

    // const [categories, setCategories] = useState(['aristocats']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories(
    //         [...categories, 'nuevaCategoria']
    //     );
    // }

    return (
        <>
        <h2>GifExpert APP</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/>

        {/* <button onClick={handleAdd}> Agregar </button> */}

            <ol>
                {categories.map( category =>
                    <GifGrid
                        key={ category }
                        categoria={ category }
                    />)
                }
            </ol>
        </>
    )
};