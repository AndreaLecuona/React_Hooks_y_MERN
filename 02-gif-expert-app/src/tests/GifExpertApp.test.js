import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('Pruebas en el GifExpertApp', () => {

    test('Debe renderizarse correctamente ', () => {
        
        const wrapper = shallow( <GifExpertApp/> ); 

        expect( wrapper ).toMatchSnapshot();

    });

    //no hay forma directa de pasarle un state por default para testear
    //por lo tanto modificamos el componenente para flexbilizarlo y darle la chance a otro usuario de pasarle un valor por defecto a categories, y por extension al useState inicial

    test('Debe mostrar una lista de categorias', () => {
        
        const categories = ['Aristocats', '101 dalmatians'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });
    
    
});