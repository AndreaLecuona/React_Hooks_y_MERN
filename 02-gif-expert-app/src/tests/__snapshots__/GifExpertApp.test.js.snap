// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpert APP
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Aristocats"
      key="Aristocats"
    />
    <GifGrid
      categoria="101 dalmatians"
      key="101 dalmatians"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el GifExpertApp Debe renderizarse correctamente  1`] = `
<Fragment>
  <h2>
    GifExpert APP
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
